iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -84,15 +84,10 @@ AMQP_TRANSPORT_STATE_BEING_DESTROYED:                  State set if IoTHubTransp
     AMQP_TRANSPORT_STATE_NOT_CONNECTED_NO_MORE_RETRIES, \
     AMQP_TRANSPORT_STATE_BEING_DESTROYED
 
-// Suppress unused function warning for AMQP_TRANSPORT_STATEstrings 
-#ifdef __APPLE__
```

Comment:
Because it was silencing the same warning, just only for __APPLE__, whereas my fix works for all platforms assuming you are using a version of c-utility more recent than [3ba539bdd74648a84972b23e99872986977f697b](https://github.com/Azure/azure-c-shared-utility/commit/3ba539bdd74648a84972b23e99872986977f697b)